### PR TITLE
fix wait_for_snapshot_purge

### PR DESCRIPTION
**Pull request by meldafrawi**
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -1217,16 +1217,17 @@ def wait_for_snapshot_purge(client, volume_name, *snaps):
     assert completed == len(purge_status)
 
     # Now that the purge has been reported to be completed, the Snapshots
-    # should be gone.
+    # should be marked as removed or exist as last snapshot.
```
Comment (yasker) [inline manager/integration/tests/common.py]: `should be removed or "marked as removed" in the cast of the latest snapshot`

File: manager/integration/tests/common.py [hunk 2]
```diff
@@ -1217,16 +1217,17 @@ def wait_for_snapshot_purge(client, volume_name, *snaps):
...
-    for snap in snapshots:
-        snap_list.append(snap["name"])
+
     for snap in snaps:
-        if snap in snap_list:
-            found = True
-            break
+        for vs in snapshots.data:
+            if snap == vs["name"]:
+                if vs["removed"] is False:
```
Comment (yasker) [inline manager/integration/tests/common.py]: The logic here is wrong. The latest snapshot (with `volume-head` as a child) needs to be marked as `removed`. All other snapshot should be removed already.

We have another case that a snapshot cannot be immediately removed, but that involves a snapshot tree.

**Comment by yasker** [comment]
LGTM.

**Comment by meldafrawi** [comment]
Changes already merged in #256

**Pull request closed without merging**